lib/builder-webpack4/src/preview/iframe-webpack.config.ts

```diff
@@ -101,7 +101,11 @@ export default async (options: Options & Record<string, any>): Promise<Configura
 
     const configEntryPath = path.resolve(path.join(workingDir, 'storybook-config-entry.js'));
     virtualModuleMapping[configEntryPath] = handlebars(
-      await readTemplate(path.join(__dirname, 'virtualModuleModernEntry.js.handlebars')),
+      await readTemplate(
+        require.resolve(
+          '@storybook/builder-webpack4/templates/virtualModuleModernEntry.js.handlebars'
```

Comment:
So here's a discussion point:

I can resolve to a file in the current package's root in (at least) 2 ways:

1. How I've done it now: refer to the whole npm package.
2. I could also opt to use `pkg-dir` with the `__dirname` of the current file.

Option 2 works by working up with the `fs` module from the specified directory until it found a `package.json` file, then return that directory.

So option 2 would look like this instead:

```ts
path.resolve(
  path.join(
    pkgDir.sync(__dirname),
    './templates/virtualModuleModernEntry.js.handlebars'
  )
)
```

In fact I HAD to use the above technique in 1 place because of unit-tests mocking the file-system.

SO should I use option 2 everywhere?

Reply:
For some reason it feels safer to me, it feels sort of weird to `require('x/...')` when we are already in package `x`. I'm not sure what the guarantees around that are.

Another alternative would be do it relative `require('../../../templates/...)`. I guess that would presuppose our `dist` folder structure and be kind of annoying.